authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -355,12 +375,21 @@ injectOAuthToCred oa cred =
             ] cred
 
 
+-- Note that prefix is used for realm in addAuthHeader, so should fix that up,
+-- but may need an API break for that.
+addAuthBody :: BS.ByteString -> Credential -> Request -> Request
+addAuthBody _ (Credential cred) req = urlEncodedBody (filterCreds cred) req
```

Comment:
I'm still not following what the motivation is, or what the question is. Why not just leave off that parameter if it's not used?